Regex concept (#125)
exercism/jq: exercises/concept/regular-chatbot/.meta/design.md

# Design

## Learning objectives

- Learn about the flavour of jq Regular Expressions.
- How to create a regular expression, in jq with strings only.
- How to use the regex related functions
    - test, match, capture, split, scan, sub, gsub
- How to pass flags
- Understanding the different parts of the match result, incl. capture results
- Using the replace method alongside regular expressions

## Out of scope

This exercise is intended as an introduction to Regular Expressions in jq.

> The main focus resides in understanding the different functions to **use** regexes

Reviewer: ```suggestion
The main focus is understanding the different functions to **use** regexes.
```
